docs/guides/js-waku/store-retrieve-messages.md

```diff
@@ -0,0 +1,219 @@
+```
+
+## Retrieve Messages
+
+`js-waku` provides the `queryOrderedCallback()` and `queryGenerator()` functions for querying `Store` nodes and retrieving historical or missed messages. The responses from `Store` nodes are paginated and require you to handle them sequentially, processing each page when received.
+
+### `queryOrderedCallback`
+
+The `store.queryOrderedCallback()` function provides a straightforward method for querying `Store` nodes and processing messages in chronological order through a callback function. It accepts these parameters:
+
+- `decoders`: List of `decoders` that specify the `content topic` to query for and their [message decryption](https://rfc.vac.dev/spec/26/) methods.
+- `callback`: The callback function for processing the retrieved messages.
+- `options` (optional): [Query options](/guides/js-waku/store-retrieve-messages#store-query-options) to filter the retrieved messages.
+
+```js
+// Create the callback function
+const callback = (wakuMessage) => {
+	// Render the message/payload in your application
+    console.log(wakuMessage);
+};
+
+// Set the query options
+const queryOptions = {
+	pageSize: 5,
+};
+
+// Query the Store node
+await node.store.queryOrderedCallback(
+	[decoder],
+	callback,
+	queryOptions,
+);
+```
+
+### `queryGenerator`
+
+The `store.queryGenerator()` function provides more control and flexibility over processing messages retrieved from `Store` nodes through [Async Generators](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/AsyncGenerator). It accepts these parameters:
+
+- `decoders`: List of `decoders` that specify the `content topic` to query for and their [message decryption](https://rfc.vac.dev/spec/26/) methods.
+- `options` (optional): [Query options](/guides/js-waku/store-retrieve-messages#store-query-options) to filter the retrieved messages.
+
+```js
+// Set the query options
+const queryOptions = {
+	pageSize: 5,
+};
+
+// Create the store query
+const storeQuery = node.store.queryGenerator(
+	[decoder],
+	queryOptions,
+);
+
+// Process the messages
+for await (const messagesPromises of storeQuery) {
+	// Fulfill all the messages promises
+    const messages = await Promise.all(messagesPromises);
+    // Render the message/payload in your application
+    console.log(messages);
+}
+```
+
+## Store Query Options
+
+### `pageDirection`
+
+The `pageDirection` option specifies the direction in which pages are retrieved:
+
+- `BACKWARD` (default): Most recent page first.
+- `FORWARD`: Oldest page first.
+
+```js
+import { PageDirection } from "@waku/sdk";
+
+// Retrieve recent messages first
+const queryOptions = {
+	pageDirection: PageDirection.BACKWARD,
+};
+
+// Retrieve oldest messages first
+const queryOptions = {
+	pageDirection: PageDirection.FORWARD,
+};
+```
+
+:::info
+The `pageDirection` option does not affect the ordering of messages within the page, as the oldest message always returns first.
```

Comment:
Actually we do re-order the messages for you if you use `queryOrderedCallback`

https://github.com/waku-org/js-waku/blob/2ceac3bdf31a7553dae55d75949a26aa0e3c0f67/packages/core/src/lib/store/index.ts#L124